### PR TITLE
55462 improve cache key site query

### PR DESCRIPTION
**Pull request by manuelRod**
Improve cache generation unsetting update_site_cache and update_site_meta_cache

Trac ticket: [https://core.trac.wordpress.org/ticket/55462](https://core.trac.wordpress.org/ticket/55462)

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/class-wp-site-query.php [hunk 1]
```diff
@@ -345,8 +345,10 @@ public function get_sites() {
 		// $args can include anything. Only use the args defined in the query_var_defaults to compute the key.
 		$_args = wp_array_slice_assoc( $this->query_vars, array_keys( $this->query_var_defaults ) );
 
-		// Ignore the $fields argument as the queried result will be the same regardless.
+		// Ignore the $fields, $update_site_cache, $update_site_meta_cache argument as the queried result will be the same regardless.
 		unset( $_args['fields'] );
+		unset( $_args['update_site_cache'] );
+		unset( $_args['update_site_meta_cache'] );
```
Comment (spacedmonkey) [inline src/wp-includes/class-wp-site-query.php]:
```suggestion
		unset( $_args['fields'], $_args['update_site_cache'], $_args['update_site_meta_cache'] );
```
You can do that

File: tests/phpunit/tests/multisite/wpSiteQuery.php [hunk 2]
```diff
@@ -863,6 +863,72 @@ public function test_wp_site_query_cache_with_same_fields_different_count() {
...
+					'fields'                 => 'ids',
+					'network_id'             => self::$network_ids['wordpress.org/'],
+					'number'                 => 3,
+					'order'                  => 'ASC',
+					'update_site_cache'      => true,
+					'update_site_meta_cache' => true,
+				)
+			);
+
+			$number_of_queries = $wpdb->num_queries;
```
Comment (spacedmonkey) [inline tests/phpunit/tests/multisite/wpSiteQuery.php]:
```suggestion
			$number_of_queries = get_num_queries();
```

File: tests/phpunit/tests/multisite/wpSiteQuery.php [hunk 3]
```diff
@@ -863,6 +863,72 @@ public function test_wp_site_query_cache_with_same_fields_different_count() {
...
+				array(
+					'fields'                 => 'ids',
+					'network_id'             => self::$network_ids['wordpress.org/'],
+					'number'                 => 3,
+					'order'                  => 'ASC',
+					'update_site_cache'      => true,
+					'update_site_meta_cache' => true,
+				)
+			);
+			$this->assertSame( $number_of_queries, $wpdb->num_queries );
```
Comment (spacedmonkey) [inline tests/phpunit/tests/multisite/wpSiteQuery.php]:
```suggestion
			$this->assertSame( $number_of_queries, get_num_queries() );
```

File: tests/phpunit/tests/multisite/wpSiteQuery.php [hunk 4]
```diff
@@ -863,6 +863,72 @@ public function test_wp_site_query_cache_with_same_fields_different_count() {
 			$this->assertSame( $number_of_queries + 1, $wpdb->num_queries );
 		}
 
+		/**
+		 * @ticket 55462
+		 */
+		public function test_wp_site_query_cache_with_same_fields_same_cache_fields() {
+			global $wpdb;
```
Comment (spacedmonkey) [inline tests/phpunit/tests/multisite/wpSiteQuery.php]:
```suggestion
```

File: tests/phpunit/tests/multisite/wpSiteQuery.php [hunk 5]
```diff
@@ -863,6 +863,72 @@ public function test_wp_site_query_cache_with_same_fields_different_count() {
...
+				)
+			);
+			$this->assertSame( $number_of_queries, $wpdb->num_queries );
+		}
+
+		/**
+		 * @ticket 55462
+		 */
+		public function test_wp_site_query_cache_with_same_fields_different_cache_fields() {
+			global $wpdb;
```
Comment (spacedmonkey) [inline tests/phpunit/tests/multisite/wpSiteQuery.php]:
```suggestion
```

File: tests/phpunit/tests/multisite/wpSiteQuery.php [hunk 6]
```diff
@@ -863,6 +863,72 @@ public function test_wp_site_query_cache_with_same_fields_different_count() {
...
+					'fields'                 => 'ids',
+					'network_id'             => self::$network_ids['wordpress.org/'],
+					'number'                 => 3,
+					'order'                  => 'ASC',
+					'update_site_cache'      => true,
+					'update_site_meta_cache' => true,
+				)
+			);
+
+			$number_of_queries = $wpdb->num_queries;
```
Comment (spacedmonkey) [inline tests/phpunit/tests/multisite/wpSiteQuery.php]:
```suggestion
			$number_of_queries = get_num_queries();
```

File: tests/phpunit/tests/multisite/wpSiteQuery.php [hunk 7]
```diff
@@ -863,6 +863,72 @@ public function test_wp_site_query_cache_with_same_fields_different_count() {
...
+				array(
+					'fields'                 => 'ids',
+					'network_id'             => self::$network_ids['wordpress.org/'],
+					'number'                 => 3,
+					'order'                  => 'ASC',
+					'update_site_cache'      => false,
+					'update_site_meta_cache' => false,
+				)
+			);
+			$this->assertSame( $number_of_queries, $wpdb->num_queries );
```
Comment (spacedmonkey) [inline tests/phpunit/tests/multisite/wpSiteQuery.php]:
```suggestion
			$this->assertSame( $number_of_queries, get_num_queries() );
```

File: tests/phpunit/tests/multisite/wpSiteQuery.php (hunk 2 above)
Comment (manuelRod) [reply tests/phpunit/tests/multisite/wpSiteQuery.php]: you are a WP encyclopedia :D

**Review by spacedmonkey — APPROVED** [review]
Good to submit to core.

**Pull request closed without merging**